apps/gamification/hooks/useTokenBalance.ts

```diff
@@ -28,8 +30,9 @@ export const useTokenBalanceByChain = (tokenAddress: Address, chainIdOverride?:
   })
 
   useEffect(() => {
-    refetch()
-  }, [blockNumber, refetch])
+    queryClient.invalidateQueries({ queryKey })
```

Comment:
After second thought, I'm a bit confused why `refetch` is changing more frequently. Shouldn't the reference to `refetch` changes whenever `queryKey` is updated? Plus react query should have its own hash function to serialize the queryKey

Reply:
Just realized `useReadContract` has this `scopeKey` option
https://wagmi.sh/react/api/hooks/useReadContract#scopekey

Can we simply add `blockNumber` to scopeKey? For example
```typescript
const { data, status, queryKey, refetch, ...rest } = useReadContract({
    chainId: chainIdOverride || chainId,
    abi: erc20Abi,
    address: tokenAddress,
    functionName: 'balanceOf',
    args: [account || '0x'],
    query: {
      enabled: !!account,
    },
    scopeKey: `useTokenBalanceByChain_${blockNumber}`,
  })
```

So there's no need for effect

Reply:
and with `placeholderData` we can keep the previous data visible